storage/innobase/row/row0import.cc

```diff
@@ -2560,6 +2583,14 @@ row_import_cfg_read_index_fields(
 		field->fixed_len = mach_read_from_4(ptr) & ((1U << 10) - 1);
 		ptr += sizeof(ib_uint32_t);
```

Comment:
How are the 32 bits of `ptr` currently allocated? (Can we avoid repeated calls to `mach_read_from_4(ptr)` too?) Could we repurpose one of those bits for "descending field"?

I see that in `row_quiesce_write_index_fields()` we are wasting 32 bits for `field->prefix_len` and `field->fixed_len`. It seems that we could simply repurpose the most significant bit (`1U << 31`) of either of those to mean "descending field". Try to pick one that when it is set, it would cause the existing version-1 format parser to fail nicely. We could also repurpose the most significant bit in the 32 bits that we write for the field name length. That length should never exceed `NAME_LEN` (192).